packages/astro/src/core/build/index.ts

```diff
@@ -230,6 +233,13 @@ class AstroBuilder {
 			cacheManifest: internals.cacheManifestUsed,
 		});
 
+		if(!hasKey && opts.settings.config.experimental.serverIslands) {
+			this.logger.info('build', `This build generated a key to encrypt props passed to Server islands. To reuse the same key across builds, set this value as ASTRO_KEY in an environment variable on your build server.
+
+ASTRO_KEY=${await encodeKey(await keyPromise)}
```

Comment:
Is it safe to print this? I'd be worried it would show in public logs

Reply:
Ah yeah good point. Hm, I'm not sure what to do then. The only other way I can think to do this is with a new command. Something like `astro generate-key` or something.

Reply:
That's probably a better way to go. Will update the PR.

Reply:
New output is like:

```
Generated a key to encrypt props passed to Server islands. To reuse the same key across builds, set this value as ASTRO_KEY in an environment variable on your build server.

ASTRO_KEY=P90X3r0+nEqzystDC1pg01VS4s/+jINyDTYSNlEO0HQ=
```